bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/ReadEntryProcessorV3.java

```diff
@@ -236,6 +236,10 @@ protected ReadResponse getReadResponse() {
         } catch (IOException e) {
             LOG.error("IOException while reading entry: {} from ledger {} ", entryId, ledgerId, e);
             return buildResponse(readResponse, StatusCode.EIO, startTimeSw);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOG.error("InterruptedException while reading entry: {} from ledger {} ", entryId, ledgerId, e);
+            return buildResponse(readResponse, StatusCode.EIO, startTimeSw);
```

Comment:
Same EIO? What can cause interrupted exception here? other than obvious kill -9 <PID> ?

Reply:
Kill -9 does not set interrupted flag on thread
This is a java artifact, it is a flag on the thread that mean 'please stop' and it is not a real signal from SO

Reply:
The interrupted exception could be caused when shutting down the bookie and, in this case, doing a fencing read. The fencing request will enqueued on the journal and the journal queue is now potentially blocking. If the read executor is being shutdown (when bookie exits) and the thread is blocked on enqueuing to the journal, it will get interrupted exception.

Previous behavior was to have always non-blocking queue that can grow indefinitely.